File: lcd-hd44780/lcd-clock.c
#include "lcd.h"
#include <stdio.h>
#include <time.h>
#include <unistd.h>

#define LCD_LINE_LENGTH  LCD_1602_LINE_LENGTH

int main()   {
    int fd_lcd; 
    char line1[LCD_LINE_LENGTH], line2[LCD_LINE_LENGTH];
    char days[7][3] = { "Son", "Mon", "Die", "Mit", "Don", "Fri", "Sam" };

    fd_lcd = i2c_setup(I2C_ADDR);
    lcd_init(fd_lcd); // setup LCD

    while(1) {
        time_t T = time(NULL);
        struct  tm tm = *localtime(&T);

        sprintf(line1, "    %02d:%02d:%02d",tm.tm_hour, tm.tm_min, tm.tm_sec);
        sprintf(line2, " %3s %02d.%02d.%04d",days[tm.tm_wday], tm.tm_mday, tm.tm_mon+1, tm.tm_year+1900);

        //lcd_clear(fd_lcd);
        lcd_write_line(fd_lcd, 1, line1);
        lcd_write_line(fd_lcd, 2, line2);
        sleep(1);
    }
    return 0;
}

File: lcd-hd44780/python/lcd_clock.py
#!/usr/bin/python
import time
from hd44780 import HD44780 as LCD

# Initialise display
lcd = LCD()

try:
    while True:
        # Send date and time
        lcd.lcd_write_line(time.strftime("    %H:%M:%S", time.localtime()), 1)
        lcd.lcd_write_line(time.strftime(" %a %d.%m.%Y ", time.localtime()), 2)
        time.sleep(1)
except KeyboardInterrupt:
    pass
finally:
    lcd.lcd_clear()
    print("Program aborted.")



